config/application.rb

```diff
@@ -8,11 +8,14 @@
 
 module ScholarsArchive
   class Application < Rails::Application
+    application_config = YAML.load_file(Rails.root.join('config/config.yml'))|| {}
```

Comment:
Can you set this to a constant like APP_CONFIG and use it in the devise initializer?